api/controllers/FileUploadController.js

```diff
@@ -61,25 +64,54 @@ const FileUploadController = {
         req.session.eApp.uploadMessages.errors = [];
         req.session.eApp.uploadMessages.fileCountError = false;
         req.session.eApp.uploadMessages.infectedFiles = [];
+        req.session.eApp.uploadMessages.noFileUploadedError = false;
     },
 
     async _errorChecksAfterUpload(req, res, err) {
-        displayErrorAndRemoveLargeFiles(req);
+        try {
+            if (req.files.length === 0) {
+                req.session.eApp.uploadMessages.noFileUploadedError = true;
+                throw new Error('No files were uploaded.');
+            }
+
+            removeLargeFiles(req);
+
+        } catch (err) {
+            sails.log.error(err);
+            FileUploadController._redirectToUploadPage(res);
+            return;
+        }
+
         if (err) {
-            const fileLimitExceeded = err.code === MULTER_FILE_COUNT_ERR_CODE;
+            const fileLimitExceeded =
+                err.code === MULTER_FILE_COUNT_ERR_CODE;
             if (fileLimitExceeded) {
                 req.session.eApp.uploadMessages.fileCountError = true;
             } else {
+                sails.log.error(err);
                 res.serverError(err);
```

Comment:
what's the difference between these?

Reply:
So to my knowledge `res.serverError` sends a 500 error to the client and redirects to the 500 error page, but maybe it logs the error as well so I don't need to log it twice 🤔 
https://sailsjs.com/documentation/reference/response-res/res-server-error

Reply:
Yep looks like it does log the error 🤦 
https://github.com/UKForeignOffice/loi-application-service/blob/develop/api/responses/serverError.js#L27